src/box/lua/console.lua

```diff
@@ -633,13 +635,13 @@ local function local_read(self)
         end
         if delim == "" then
             local lang = self.language
-            if not lang or lang == 'lua' then
+            if ((not lang or lang == 'lua') and local_check_lua(buf)) or (lang and lang ~= 'lua') then
```

Comment:
This code is cryptic in my opinion. I would propose to make it more linear and accompany with explanation how it works.

```lua
local lang = self.language or 'lua'
if lang == 'lua' then
    -- We should stop reading input, when it is valid Lua
    -- statement or expression or when it is syntactically
    -- incorrect (not due to incompleteness).
    local is_complete = local_check_buf(buf)
    -- Interpret backslash as a line continuation marker.
    --
    -- However we don't do that for a backlash at end of a line
    -- within a string due to backward compatibility concerns.
    -- is_complete is false in this case and there are no other
    -- such cases according to Lua syntax.
    if is_complete and line:endswith(multiline_delim) then
        is_complete = false
        buf = buf:sub(1, -1 - #multiline_delim)
    end
```

However I think more on this and discussed with @igormunkin and now I think that it would be better to don't hack around a backslash at end of a line within a string, but rather interpret backslash under and option like `\set continuation true|false` (default is false).

Reply:
Reworked the patch, now user can set continuation char by `\set continuation \`, by default it is off.

Reply:
I still think that current code is cryptic and still wait for Igor's opinion on it.

Reply:
Discussion continued below.